### PR TITLE
test(desktop-tauri): add comprehensive command coverage for git status and hook trust flows

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
Add deep command-level test coverage for the two high-risk Tauri command paths identified in the code quality audit:
- `git_get_worktree_status` in `apps/desktop/src-tauri/src/commands/git.rs`
- `workspace_set_trusted_hooks` in `apps/desktop/src-tauri/src/commands/workspace.rs`

This closes gaps where only helper-level tests existed and command-level behavior (authorization, worktree resolution, challenge validation/replay handling, and IPC invocation boundaries) was not directly protected.

### 1) `git_get_worktree_status` command-level coverage
File: `apps/desktop/src-tauri/src/commands/git.rs`

Added a command-focused test harness using `tauri::test` + managed `AppState` + a controllable fake `GitPort`, then added tests for:
- unauthorized repository rejection at command boundary
- upstream-status error propagation from port failures
- preservation of upstream error variant in aggregated payload + snapshot metadata assertions
- rejection of unrelated `working_dir` (authorization/linked-worktree checks)
- acceptance of registered worktree `working_dir` and verification of effective path forwarded to `GitPort`

Also adjusted the fixture setup to initialize a real git repo so command-level `working_dir` resolution paths are exercised end-to-end.

### 2) `workspace_set_trusted_hooks` command-level coverage
File: `apps/desktop/src-tauri/src/commands/workspace.rs`

Extended command-level tests to cover challenge and trust flow edge cases:
- missing nonce/fingerprint rejection
- expired challenge rejection
- fingerprint mismatch rejection
- repository mismatch rejection
- nonce replay rejection after first consumption
- stale challenge rejection when hooks mutate after challenge generation
- disable path (`trusted=false`) without challenge data

### 3) IPC invocation boundary test
File: `apps/desktop/src-tauri/src/commands/workspace.rs`

Added an explicit IPC-level test (using `tauri::test::get_ipc_response`) for `workspace_set_trusted_hooks` on a dialog-free error branch to validate command handler registration and invoke argument mapping, not only direct function invocation.

### 4) Runtime-generic command intent documentation
File: `apps/desktop/src-tauri/src/commands/workspace.rs`

Added concise comments clarifying why runtime-generic signatures are used for `workspace_save_repo_settings`, `workspace_set_trusted_hooks`, and `confirm_hook_trust_dialog` (MockRuntime testability without behavior changes under Wry).

### 5) Test dependency support
File: `apps/desktop/src-tauri/Cargo.toml`

Added dev dependency feature support for Tauri test utilities:
- `tauri = { version = "2", features = ["test"] }`

## Why
These command handlers contain multi-branch logic with high regression risk (authorization, canonicalization/worktree checks, challenge nonce/fingerprint/expiry semantics, replay behavior). This PR ensures those contracts are directly tested at command level instead of relying only on helper tests.

## Validation
Executed focused Rust test suites in `apps/desktop/src-tauri`:
- `cargo test --lib git_get_worktree_status_`
- `cargo test --lib workspace_set_trusted_hooks_`
- `cargo test --lib commands::git::tests::`
- `cargo test --lib commands::workspace::tests::`

All passed.

## Risk / Compatibility
- No production workflow logic change intended beyond testability-oriented signature generalization across Tauri runtimes.
- Existing command behavior and error messages are preserved; this PR primarily increases behavioral test coverage and lock-in of command contracts.

### CODE REVIEW

File: apps/desktop/src-tauri/src/commands/workspace.rs [hunk 1]
```diff
@@ -397,9 +403,388 @@ fn normalize_hook_commands(commands: &mut Vec<String>) {
...
+        assert_eq!(updated.path, canonical_repo_key(fixture.repo_path.as_str()));
+
+        let repo_config = fixture
+            .service
+            .workspace_get_repo_config(fixture.repo_path.as_str())
+            .map_err(|error| error.to_string())?;
+        assert!(!repo_config.trusted_hooks);
+        assert!(repo_config.trusted_hooks_fingerprint.is_none());
+        Ok(())
+    }
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src-tauri/src/commands/workspace.rs:859]: Implemented in 43e8132.

Added full command-level happy-path coverage for `workspace_set_trusted_hooks` via `workspace_set_trusted_hooks_accepts_valid_challenge_and_persists_trust`.

What is now covered end-to-end:
- valid nonce/fingerprint challenge setup,
- trust confirmation path (test-controlled dialog acceptance),
- persistence assertions for `trusted_hooks=true` and matching stored fingerprint.

To keep production behavior unchanged while making this path testable, the dialog response override is compiled only under `#[cfg(test)]` on `AppState` and consumed only in test builds.

**Comment by Maxsky5** [comment]
Reviewed all PR feedback and triaged each item:

- `https://github.com/Maxsky5/openducktor/pull/154#discussion_r2881000603`: fixed in `43e8132` (happy-path trusted-hooks command test with dialog confirmation + persisted trust assertions).
- `https://github.com/Maxsky5/openducktor/pull/154#issuecomment-3994186013` (Gemini summary): informational summary only, no actionable delta; no code change required.
- `https://github.com/Maxsky5/openducktor/pull/154#issuecomment-3994193205` (Greptile summary): informational summary only, no actionable delta; no code change required.
- `https://github.com/Maxsky5/openducktor/pull/154#pullrequestreview-3885678290`: duplicated the inline happy-path suggestion above, now addressed by `43e8132`.

Validation rerun after changes:
- `bun run build`
- `bun run test`
- `bun run typecheck`
- `bun run lint`
- `cd apps/desktop/src-tauri && cargo test`

All passed.
